src/wallet/rpcwallet.cpp

```diff
@@ -537,6 +538,89 @@ static RPCHelpMan sendtoaddress()
     };
 }
 
+static RPCHelpMan walletestimatefee()
+{
+    return RPCHelpMan{"walletestimatefee",
+        "\nEstimates the fee per kilobyte needed for a transaction to be generated from this wallet,\n"
+        "taking into account mempool minfee and fallback fee.\n"
+        "In contrast, estimatesmartfee estimates ignores these and returns a pure fee estimate based on prior block confirmations.\n"
+        "Confirmation within conf_target blocks if possible and return the number of blocks\n"
+        "for which the estimate is valid.\n",
+        {
+            {"conf_target", RPCArg::Type::NUM, RPCArg::Optional::NO, RPCArg::DefaultHint{"wallet -txconfirmtarget"}, "Confirmation target in blocks (1 - 1008)"},
+            {"estimate_mode", RPCArg::Type::STR, RPCArg::Default{"conservative"}, "The fee estimate mode.\n"
+            "                   Whether to return a more conservative estimate which also satisfies\n"
+            "                   a longer history. A conservative estimate potentially returns a\n"
+            "                   higher feerate and is more likely to be sufficient for the desired\n"
+            "                   target, but is not as responsive to short term drops in the\n"
+            "                   prevailing fee market. Must be one of (case insensitive):\n"
+            "\"" + FeeModes("\"\n\"") + "\""},
+                },
+                RPCResult{
+                    RPCResult::Type::OBJ, "", "",
+                    {
+                        {RPCResult::Type::NUM, "feerate", /* optional */ true, "estimate fee rate in " + CURRENCY_UNIT + "/kvB (only present if no errors were encountered)"},
```

Comment:
We're in the process of transitioning from BTC/kvB to sat/vB for fee rates, and of no longer referring to fee rates as fees.  For example, it was planned in December 2020 to soft-replace the settxfee and estimatesmartfee calls with sat/vB equivalents named setfeerate (#20391) and estimatefeerate; the BTC/kvB ones would become hidden RPCs to not break user space.

So, if we do add another fee rate RPC, consider making it denominated in sat/vB units and referring to "fee rates", rather than "fees".

Reply:
See https://github.com/bitcoin/bitcoin/pull/20484#issuecomment-734786305 for more info.